### PR TITLE
fix: validation token

### PR DESCRIPTION
**Pull request by JoowonYun**
## Summary of changes

<Describe summary of major changes here>

## Report of required housekeeping

- [x] Github issue OR spec proposal link
- [x] Wrote tests
- [ ] Added a relevant changelog entry to CHANGELOG.md

----

## (FOR ADMIN) Before merging

- [ ] Added appropriate labels to PR
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" (coding standards)
- [ ] Confirm added tests are consistent with the intended behavior of changes
- [ ] Ensure all tests pass

### CODE REVIEW

File: packages/terraswap/src/mock_querier.rs [hunk 1]
```diff
@@ -122,6 +141,40 @@ impl Querier for WasmMockQuerier {
...
+    }
+}
+
+#[derive(Clone, Default)]
+pub struct IbcQuerier {
+    denom_traces: HashMap<String, DenomTrace>,
+}
+
+impl IbcQuerier {
+    pub fn new(denom_treaces: &[(&String, (&String, &String))]) -> Self {
```
Comment (jhlee-young) [inline packages/terraswap/src/mock_querier.rs:163]: typo denom_treaces -> denom_traces

**Review by YunSuk-Yeo — APPROVED** [review]

File: packages/terraswap/src/mock_querier.rs [hunk 2]
```diff
@@ -146,12 +199,25 @@ impl WasmMockQuerier {
                         }
                         _ => panic!("DO NOT ENTER HERE"),
                     }
+                } else if route == &TerraRoute::Market {
```
[the remark this answers is not in the document]
Comment (JoowonYun) [reply packages/terraswap/src/mock_querier.rs]: https://github.com/terraswap/terraswap/pull/39/commits/782ddc5955fb7f5c46b8a4723a6f82d20b6fdc57

File: packages/terraswap/src/mock_querier.rs (hunk 1 above)
Comment (JoowonYun) [reply packages/terraswap/src/mock_querier.rs]: https://github.com/terraswap/terraswap/pull/39/commits/782ddc5955fb7f5c46b8a4723a6f82d20b6fdc57

File: packages/terraswap/src/asset.rs [hunk 3]
```diff
@@ -196,6 +198,34 @@ impl AssetInfo {
...
+                    let res_actives = query_active_denoms(querier);
+                    if res_actives.is_err() {
+                        return false;
+                    }
+
+                    let actives = res_actives.unwrap();
+                    let res = actives.iter().find(|active_denom| *active_denom == denom);
+                    if res != None {
+                        return true;
+                    }
```
[the remark this answers is not in the document]
Comment (JoowonYun) [reply packages/terraswap/src/asset.rs]: https://github.com/terraswap/terraswap/pull/39/commits/782ddc5955fb7f5c46b8a4723a6f82d20b6fdc57

File: packages/terraswap/src/asset.rs [hunk 4]
```diff
@@ -196,6 +198,34 @@ impl AssetInfo {
             }
         }
     }
+
+    pub fn is_validate(&self, querier: &QuerierWrapper) -> bool {
```
[the remark this answers is not in the document]
Comment (JoowonYun) [reply packages/terraswap/src/asset.rs]: https://github.com/terraswap/terraswap/pull/39/commits/782ddc5955fb7f5c46b8a4723a6f82d20b6fdc57

**Review by jhlee-young — APPROVED** [review]

**Comment by JoowonYun** [comment]
Avoid conflicts, rebase on `next`

**Pull request merged**